extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -72,79 +73,93 @@ public Uni<SecurityIdentity> authenticate(TokenAuthenticationRequest request,
             return Uni.createFrom().nullItem();
         }
         LOG.debug("Starting creating SecurityIdentity");
-        RoutingContext vertxContext = HttpSecurityUtils.getRoutingContextAttribute(request);
-        vertxContext.put(AuthenticationRequestContext.class.getName(), context);
 
-        Uni<TenantConfigContext> tenantConfigContext = tenantResolver.resolveContext(vertxContext);
-
-        return tenantConfigContext.onItem()
+        return resolveTenantConfigContext(request, context).onItem()
                 .transformToUni(new Function<TenantConfigContext, Uni<? extends SecurityIdentity>>() {
                     @Override
                     public Uni<SecurityIdentity> apply(TenantConfigContext tenantConfigContext) {
                         return Uni.createFrom().deferred(new Supplier<Uni<? extends SecurityIdentity>>() {
                             @Override
                             public Uni<SecurityIdentity> get() {
-                                return authenticate(request, vertxContext, tenantConfigContext);
+                                return authenticate(request, getRequestData(request), tenantConfigContext);
                             }
                         });
                     }
                 });
     }
 
-    private Uni<SecurityIdentity> authenticate(TokenAuthenticationRequest request,
-            RoutingContext vertxContext,
+    public Uni<SecurityIdentity> authenticate(String accessToken) {
+        return authenticate(new TokenAuthenticationRequest(new AccessTokenCredential(accessToken)),
+                new AuthenticationRequestContext() {
+                    @Override
+                    public Uni<SecurityIdentity> runBlocking(Supplier<SecurityIdentity> function) {
+                        return Arc.container().instance(BlockingSecurityExecutor.class).get().executeBlocking(function);
```

Comment:
I think once you introduce an interface you'd be able to move this code out of the identity provider, and also, not run it necessarily with the blocking executor

Reply:
> I think once you introduce an interface you'd be able to move this code out of the identity provider, and also, not run it necessarily with the blocking executor

yes, I agree I'll move this code out, but the blocking executor needs to be created because it is passed to the identity provider. I'm not using it, just complying the interface. anyway, it will be out

Reply:
so, I've moved it to the `TenantIdentityProvider`. Please note that blocker execution is not there because I'm using it, but because the `SecurityIdentity` established contract.